tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+    padding: 6px;
+    width: 100%;
+    height: 100%;
+}
+
+.inner {
+    background-color: hsl(210deg 36.67% 11.76%);;
+    width: 100%;
+    height: 100%;
+}
+
+.form {
+    display: flex;
+    font-family: sans-serif;
+    flex-direction: column;
+    justify-content: space-evenly;
+    padding: 0 32px;
+    width: 100%;
+    height: 100%;
+}
+
+input {
+    all: unset;
+    background: hsl(218, 80%, 2%);
+    color: white;
+    cursor: text;
+    font-size: 1rem;
+    height: 56px; 
+    outline-color: hsl(355, 100%, 63%);
+    outline-width: thin;
+    padding: 0 32px;
+}
+
+input::placeholder {
+    color: hsl(0, 0%, 50%);
+}
+
+input:focus-visible {
+    outline-style: inset;
+}
+
+button {
+    background-color: hsl(218, 80%, 2%);
+    border: unset;
+    cursor: pointer;
+    font-family: sans-serif;
+    font-size: 1rem;
+    font-weight: 600;
+    height: 56px;
+    outline-color: hsl(355, 100%, 63%);
+    outline-width: thin;
+}
+
+button:focus-visible {
+    outline-style: inset;
+}
+
+button span {
+    background: linear-gradient(92.05deg, hsl(355, 100%, 68%) 12.09%, hsl(355, 100%, 63%) 42.58%, hsl(355, 85%, 54%) 84.96%);
+    -webkit-background-clip: text;
+    background-clip: text; 
+    -webkit-text-fill-color: hsl(0deg 0% 0% / 0%);
+    font-size: 1.2rem;
+}
+</code></pre>
+
+        <div class="paper" id="paper-foreign-object-prevent-default"></div>
+        <p>JointJS source code: <a href="js/foreign-object.js" target="_blank">foreign-object.js</a></p>
+
+
+        <p>
+            The joy of working with <code>foreignObject</code> in our shapes is that we still get to take advantage of native web functionality.
+            When using a <code>form</code>, you might be used to programmatically associating a <code>label</code> with an <code>input</code> 
+            by providing an <code>'id'</code>
+            attribute to the <code>input</code>, and a <code>'for'</code> attribute to the <code>label</code> which has the same value. This has
+            a number of advantages such as screen readers reading the label when the associated input is focused, and also that the focus is 
+            passed to the input when the associated label is clicked.
+        </p>
+
+        <p>
+            On the other hand, when working with JointJS, you will have to make decisions about how some HTML elements lose focus. If a user clicks 
+            on a blank area, you may wish that an <code>input</code> lose focus which is behavior we are familiar with from HTML. In order to mimic 
+            this behaviour in JointJS, we can create an event to remove focus when the user clicks on a blank <code>paper</code> area, or even on
+            any JointJS element.
+        </p>
+
+        <pre><code>paper.on('blank:pointerdown cell:pointerdown', () => {
+    document.activeElement.blur();
+});
+</code></pre>
+
+        <p>
```

Comment:
It feels like a new section is starting here... the following paragraphs feel disconnected from everything that came before in this section "A more interactive example". Can you come up with a heading for the following paragraphs?

Reply:
"Preventing default JointJS and browser behavior" maybe?